src/linear_algebra/smodeq.lean

```diff
@@ -14,66 +15,99 @@ import ring_theory.ideal.quotient
 open submodule
 open_locale polynomial
 
-variables {R : Type*} [ring R]
-variables {M : Type*} [add_comm_group M] [module R M] (U U₁ U₂ : submodule R M)
+variables {M X : Type*} [add_comm_group M] [set_like X M] [add_subgroup_class X M] (U U₁ U₂ : X)
+variables {R : Type*} [ring R] [module R M] (Ug : add_subgroup M) (Us : submodule R M)
```

Comment:
From a lean perspective this isn't really a generalization. Can you use `add_subgroup_class` instead, so that it can still be used with submodules?